tests/system/src/test/java/test/javafx/scene/control/ProgressIndicatorLeakTest.java

```diff
@@ -38,57 +39,83 @@
 import java.util.LinkedList;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
 
 import junit.framework.Assert;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
 import org.junit.Test;
+import static org.junit.Assert.assertTrue;
+import static test.util.Util.TIMEOUT;
+import test.util.Util;
 import test.util.memory.JMemoryBuddy;
 
 public class ProgressIndicatorLeakTest {
 
-    static CountDownLatch startupLatch;
-    static WeakReference<Node> detIndicator = null;
-    static Stage stage = null;
+    @BeforeClass
+    public static void initFX() throws Exception {
+        CountDownLatch startupLatch = new CountDownLatch(1);
+        Platform.setImplicitExit(false);
+        Platform.startup(startupLatch::countDown);
+        Assert.assertTrue("Timeout waiting for FX runtime to start",
+                startupLatch.await(TIMEOUT, TimeUnit.MILLISECONDS));
+    }
 
-    public static class TestApp extends Application {
+    @Test
+    public void leakDeterminationIndicator() throws Exception {
+        JMemoryBuddy.memoryTest((checker) -> {
+            Util.runAndWait(() -> {
+                Stage stage = new Stage();
+                ProgressIndicator indicator = new ProgressIndicator(-1);
+                indicator.setSkin(new ProgressIndicatorSkin(indicator));
+                Scene scene = new Scene(indicator);
+                stage.setScene(scene);
+                indicator.setProgress(1.0);
+                Assert.assertEquals("size is wrong", 1, indicator.getChildrenUnmodifiable().size());
+                Node detIndicator = indicator.getChildrenUnmodifiable().get(0);
+                indicator.setProgress(-1.0);
+                indicator.setProgress(1.0);
+                checker.assertCollectable(detIndicator);
+                stage.show();
```

Comment:
That's a good question.
OnShown is usually called in about 1 Frame.
The memory-leak test takes some times, usually longer as 1 Frame. 
In the current configuration, it takes up to 1 seconds, with up to 10 checks.
Because the check is so much longer, waiting for onShown doesn't really make a difference.

Reply:
My worry isn't the test might fail and give a false positive. The checking 10 times will be sufficient for that.

My concern is that since this bug fix is fixing a leak that only happens when the stage is showing, the test might miss a potential leak if a regression in this area were ever reintroduced. I think it's best to restore the wait for onShown.

Reply:
I've readded the wait onShown. I don't really see a Scenario were this makes a difference, but i guess it's more correct, because it now tests it at the moment, where the object should be free, and not relying on the fact, that the check takes some time.